lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -134,200 +135,136 @@ import {
-
-				flatpickr = flatpickrFunction;
-			}
-
-			const { Polish } = await import('flatpickr/dist/l10n/pl.js');
-
-			this.datePicker = flatpickr(this.$refs.flatpickrInput, {
-				locale: Polish,
-				positionElement: this.$el,
-				ignoredFocusElements: [this.$el],
-				appendTo: this.$el,
-				position: FLATPICKR_POSITIONS[this.calendarPosition],
-				defaultDate: this.date,
-				disable: this.disableDates.filter((date) => date instanceof Date),
-				minDate: this.minDate,
-				maxDate: this.maxDate,
-				onClose: [
-					() => {
-						this.isOpen = false;
-					},
-				],
-				onOpen: [
-					() => {
-						this.isOpen = true;
-					},
-				],
-				onChange: (event) => {
-					this.$emit('update:date', event[0]);
-				},
-			});
-		}
+	isIconHiddenOnMobile: {
+		type: Boolean,
+		default: false,
 	},
-	unmounted() {
-		this.datePicker?.destroy();
-		this.datePicker = null;
+	errorMessage: {
+		type: String,
+		default: '',
 	},
-	methods: {
-		toggle() {
-			if (this.isDatePickerEnabled && this.datePicker) {
-				this.datePicker.toggle();
-			}
-		},
+	state: {
+		type: String as PropType<DatePickerStates>,
+		default: DATE_PICKER_STATES.DEFAULT,
+	},
+	color: {
+		type: String as PropType<DatePickerColors>,
+		default: DATE_PICKER_COLORS.NEUTRAL,
+	},
+	calendarPosition: {
+		type: String as PropType<DatePickerCalendarPositions>,
+		default: DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT,
+	},
+	disableDates: {
+		type: Array as PropType<Array<Date>>,
+		default: () => [],
+	},
+	minDate: {
+		type: Date,
+		default: null,
+	},
+	maxDate: {
+		type: Date,
+		default: null,
 	},
 });
+
+const emit = defineEmits({ 'update:date': (value: Date) => true });
+const onChange = (event: Array<Date>) => {
+	emit('update:date', event[0]);
+};
+
+const eyebrowText = computed(() => {
+	if (!props.date || props.state === DATE_PICKER_STATES.LOADING) {
+		return '';
+	}
+
+	return capitalizeFirstLetter(localWeekdayName(props.date));
+});
+
+const text = computed(() => {
+	if (props.state === DATE_PICKER_STATES.LOADING || !props.date) {
+		return props.placeholder;
+	}
+
+	return localFullDateWithShortMonthName(props.date);
+});
+
+const {
+	isOpen,
+	toggle: toggleDatePicker,
+	createDatePicker,
+} = initFlatpickr({
+	flatpickrInputRef,
+	dateRangePickerRef,
+	props,
+	onChange,
+	defaultDates: props.date,
+});
+function toggle() {
+	if (props.isInteractive && props.state === DATE_PICKER_STATES.DEFAULT) {
+		toggleDatePicker();
+	}
+}
+
+onMounted(async () => {
+	if (props.isInteractive && props.state === DATE_PICKER_STATES.DEFAULT) {
+		await createDatePicker();
+	}
+});
+
+watch([() => props.isInteractive, () => props.state], async () => {
+	if (props.isInteractive && props.state === DATE_PICKER_STATES.DEFAULT) {
+		await createDatePicker();
+	}
+});
+
+const tileIcon = computed(() => {
+	if (!props.icon) {
+		return null;
+	}
+
+	return isOpen.value ? ICONS.FA_CHEVRON_UP : props.icon;
+});
```

Comment:
IMO: we should definitely have some linterns for composition files - it's a terrible mess - I will add this on retro

Reply:
We should propose something for retro. As far as I know Igor and Damian will ask about what we introduce since we are the first creating component in composition api

Reply:
We can start with https://eslint.vuejs.org/rules/

Reply:
Sorry, it is for options api